### DISCUSSION TITLE
Which section does this belong to? Added, Fixed or Changed

### DISCUSSION THREAD

**Discussion by ayub-kokabi** [olivierlacan/keep-a-changelog#569] [Q&A]
I replaced three parameters of a function with one parameter. I added a new internal function to support this changes. I improved logic and some fixes for this new parameter that affect performance. Which section does this belong to? Added, fixed, or changed?

One of the **main challenges** I face when using the KeepChangelog guideline is determining which section each change belongs to.

**Comment by jagerber48** [comment]
This looks like `Changed` since you changed how an existing feature works. You may also want to include a **breaking** flag if your changes broke the previous API.

**Reply by fugerit79** [reply]
@sir-kokabi 

I guess there is some degree of personal preference about it. This is a question I often ask myself too.
In my opinion is to be consistent and maybe state the criteria used.

I would try to explain by example. 
Let's say in our porject revolve around a structured object, shapes, something like this in version 1.0.0 : 

| **shape** |
|------|
| - height |
| - width |

In version 1.1.0 we add "depth" property.

This can be 

## Added

- depth property to 'shape'

Or it can be

## Changed

- now 'shape' accepts depth property too

And following modification are consistent to this decision. (for instance "add" for new objects and "change" for modified objects).

Does it make sense?

**Reply by firefoxic** [reply]
> Or it can be
> ### Changed
>
> - now 'shape' accepts depth property too

This doesn't seem right to me. The important thing here is not that the object has changed by adding a new property, but that the property has been added and the rest of the API has not been changed, and the user, who is satisfied with the previous functionality, will not have to change anything in his code. So this minor change better be in `Added` and when publishing a new version of the package in the absence of the `Changed` section, the minor (second) part of the version number should be raised.

**Reply by fugerit79** [reply]
@firefoxic I was not saying first or second proposal are right or wrong.
I was just trying to point out that sometimes, there is not only one possible criteria to categorize the change set.
But one you pick a criteria for your project you should be consistent in each version in my opinion.
That's all.

Honestly talking about the specific example was not my point.

Thanks for the feedback anyway :)
